src/context/CartContext.tsx

```diff
@@ -0,0 +1,155 @@
+  cartTotalPrice: number
+}
+
+export const CartContext = createContext<CartContextType>({
+  cart: [],
+  addToCart: (product: TypeCartProduct) => {},
+  deleteFromCart: (title: string) => {},
+  decrementFromCart: (title: string) => {},
+  incrementToCart: (title: string) => {},
+  navLinks: [],
+  cartQuantity: 0,
+  cartTotalPrice: 0,
+})
+
+export default function CartProvider({ children }: { children: ReactNode }) {
+  const [navLinks, setNavLinks] = useState<string[]>([])
+
+  useEffect(() => {
+    const getLinks = async () => {
+      try {
+        const storedNavLinks = localStorage.getItem('navLinks')
+        if (storedNavLinks) {
+          setNavLinks(JSON.parse(storedNavLinks))
+        } else {
+          const data = await fetchGql(queryForNavLinks)
+          const fetchedLinks = data.navLinks.links
+          localStorage.setItem('navLinks', JSON.stringify(fetchedLinks))
+          setNavLinks(fetchedLinks)
+        }
+      } catch (err) {
+        throw new Error(`Error fetching data: ${err}`)
+      }
+    }
+
+    getLinks()
+  }, [])
+
+  const [cart, setCart] = useState<TypeCartProduct[]>([])
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const storedCart = localStorage.getItem('cart')
+      if (storedCart) {
+        setCart(JSON.parse(storedCart))
+      }
+    }
+  }, [])
+
+  const addToCart = useCallback((product: TypeCartProduct) => {
+    setCart((prevCart) => {
+      const index = prevCart.findIndex((item) => item.title === product.title)
+
+      if (index !== -1) {
+        return prevCart.map((item, i) =>
+          i === index
+            ? { ...item, quantity: item.quantity + product.quantity }
+            : item
+        )
+      }
+      if (typeof window !== 'undefined') {
+        window.localStorage.setItem(
+          'cart',
+          JSON.stringify([...prevCart, product])
+        )
+      }
+      return [...prevCart, product]
+    })
+  }, [])
+
+  const deleteFromCart = useCallback((title: string) => {
+    setCart((prevCart) => {
+      const newCart = prevCart.filter((product) => product.title !== title)
+      localStorage.setItem('cart', JSON.stringify(newCart))
+      return newCart
+    })
+  }, [])
+
+  const decrementFromCart = useCallback((title: string) => {
+    setCart((prevCart) => {
+      const newCart = prevCart
+        .map((item) =>
+          item.title === title ? { ...item, quantity: item.quantity - 1 } : item
+        )
+        .filter((item) => item.quantity > 0)
+      localStorage.setItem('cart', JSON.stringify(newCart))
+      return newCart
+    })
+  }, [])
+
+  const incrementToCart = useCallback((title: string) => {
+    setCart((prevCart) => {
+      const newCart = prevCart.map((item) =>
+        item.title === title ? { ...item, quantity: item.quantity + 1 } : item
+      )
+      localStorage.setItem('cart', JSON.stringify(newCart))
+      return newCart
+    })
+  }, [])
+
+  const cartTotalPrice = useMemo(() => {
```

Comment:
do we need the `useMemo` here? since when we passed it we already wrapped it in a `useMemo`

Reply:
removed useMemo for cartTotalPrice and cartQuantity